content/2019/personal-notetaking-in-vim/index.md

```diff
@@ -0,0 +1,343 @@
+
+
+### Sidenote: Create + Consume in Vim
+
+Yes, you did see non-Vim tools in my list of previously used tools for notes...
+It should be noted that I was using Vim as my main editor long before I
+switched to using Vim to take notes. The reason why is that for a time I knew I
+wanted to create and edit notes using Vim, but I wanted to be able to read
+those notes in other ways, such as rich text from rendered markdown
+(Simplenote), or a wiki that was navigable in a browser (Vimwiki). The times I
+used Vim to edit was frustrating, because I needed to fit with a workflow that
+required rebuilding a wiki after editing, or opening one file at a time for
+something like Simplenote.
+
+Finally, I realized that I didn't actually need to read rich text or navigate
+hyperlinks with a mouse in a browser. I could create *and consume* in Vim!
+Granted, it's not as pretty for viewing sometimes, but now creating, editing,
+searching, and reading notes are all the same thing. Efficiency.
+
+
+### Zettelkasten
+
+One more thing to discuss before we dive in - what is Zettelkasten?.
+Zettelkasten (German for "card index") is a method for personal knowledge
+management in which one uses many small atomic notes, linked to other such
+notes. The idea is that it forms a huge interconnected network of notes one can
+traverse and interact with. I came across this methodology a little while back
+and love the idea. I haven't yet spent the time to really learn how to use it
+effectively, but I've started to use some of the ideas, including linking
+between notes and facilitating easy creation of notes.
+
+If you're interested to know more about Zettelkasten as a system, see
+[zettelkasten.de][zettelkasten] and/or web search for it. There are many good
+resources. What I described above is a horrendous simplification.
+
+
+## Let's get started!
+
+Ok, so now we have a methodology and Vim without any specific notes/wiki
+plugins. What do we do now? Let's work out the workflows involved. So we want:
+
+- Easy creation of new notes. It should be frictionless to create and start
+  editing a new note at any time. More friction equals less motivation to write
+  up a note.
+- Powerful options for search. Zettelkasten eschews hierarchy and taxonomy in
+  favour of flexible search and...
+- linking between notes. We need to be able to create a network of small notes,
+  where we can search to find an entry point, and then traverse notes to
+  discover related ideas.
+
+### Creating notes
+
+Obviously I'll be editing a new note in Vim. There are two main places from
+where I want to be able to create a note: the shell and Vim itself. It must be
+as frictionless as possible to create new notes; any friction will dissuade me
+from taking a note at once, and thoughts are fleeting.
+
+So, from inside Vim, I have a command and function to create a timestamped file
+in my notes directory:
+
+```vim
+" .vim/plugin/local.vim
+command! -nargs=* Zet call local#zettel#edit(<f-args>)
```

Comment:
Maybe beyond the scope of what you're trying to say in the article, but I would be tempted to pass the `<mods>` here too, so that `:botright vertical Zet a quick note` works as you would expect.

Reply:
Though I like this, in my workflow I almost never want this to open a new split - I always want it to replace the current one. If I really do, then I create a split first and then open it. It could be nice to mention, but https://vimways.org/2019/writing-vim-plugin/ already covers `<mods>`, so I guess I'll leave it out.